### PR TITLE
Fix gcs walker relative key for import

### LINKED ISSUE #5102: [Bug]: Importing files from Google Cloud Storage creates an empty folder if trailing slash is not used with path
### What happened?

Current Behavior:
Importing files from Google Cloud Storage creates an empty folder if trailing slash is not used with the path. It works fine when importing from S3.
        
Steps to Reproduce:
1. See the images to reproduce the issue.

image (1)

image

image (2)

image (3)

### Expected Behavior

Expected Behavior:
How you expected thing to work.

### lakeFS Version

0.90.1

### Deplyoment

GCP

### Affected Clients

_No response_

### Relevant logs output

_No response_

### Contact Details

amit.kesarwani@treeverse.io

### LINKED ISSUE #5104: [Bug]: Can not import a single file
### What happened?

Current Behavior:
Want to import a single file (e.g. gs://lakefs-amit-sample-dog-images/n02085620-Chihuahua/n02085620_10074.jpg) but can't import it via UI.

Steps to Reproduce:
1. See 1st screenshot. Import From: gs://lakefs-amit-sample-dog-images/n02085620-Chihuahua/n02085620_10074.jpg
2. See 2nd screenshot: I get the message that "1 objects imported"
3. See 3rd screenshot: But there is no object under imported branch.
4. See 4th screenshot: comparing branches also don't show any changes.
5. See 5th screenshot: Received an error when imported a single file from S3
 
Screen Shot 2023-01-25 at 5 10 53 PM

Screen Shot 2023-01-25 at 5 11 23 PM

Screen Shot 2023-01-25 at 5 11 47 PM

Screen Shot 2023-01-25 at 5 21 59 PM

Screen Shot 2023-01-25 at 5 24 31 PM

### Expected Behavior

Expected Behavior:
Should be able to import a single file.

### lakeFS Version

0.90.1

### Deplyoment

GCP

### Affected Clients

N/A

### Relevant logs output

_No response_

### Contact Details

amit.kesarwani@treeverse.io

### LINKED ISSUE #5113: [Bug]: Can't import single file by using Python client
### What happened?

Current Behavior:
Importing a single file from Google Cloud Storage via Python client creates empty file.
Facing same problem from the UI also and opened another issue for that: https://github.com/treeverse/lakeFS/issues/5104
       
Steps to Reproduce:
1. See the attached Python code to reproduce this issue

[Import 1 File.txt](https://github.com/treeverse/lakeFS/files/10510755/Import.1.File.txt)

### Expected Behavior

Expected Behavior:
Should be able to import single file.

### lakeFS Version

0.90.1

### Deplyoment

GCP

### Affected Clients

python lakefs-client 0.90.1

### Relevant logs output

_No response_

### Contact Details

amit.kesarwani@treeverse.io

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #5104
Fix #5102
Fix #5113

### CODE REVIEW

File: pkg/ingest/store/gcs.go [hunk 1]
```diff
@@ -23,6 +23,10 @@ func NewGCSWalker(client *storage.Client) *gcsWalker {
 
 func (w *gcsWalker) Walk(ctx context.Context, storageURI *url.URL, op WalkOptions, walkFn func(e ObjectStoreEntry) error) error {
 	prefix := strings.TrimLeft(storageURI.Path, "/")
+	var basePath string
+	if idx := strings.LastIndex(prefix, "/"); idx != -1 {
+		basePath = prefix[:idx+1]
```
Comment (itaiad200) [inline pkg/ingest/store/gcs.go:28]: Is this what we want? I asked to import `gs://foo/bar` and you'll import `gs://foo/`

File: pkg/ingest/store/gcs.go [hunk 2]
```diff
@@ -51,7 +54,7 @@ func (w *gcsWalker) Walk(ctx context.Context, storageURI *url.URL, op WalkOption
 		}
 		if err := walkFn(ObjectStoreEntry{
 			FullKey:     attrs.Name,
-			RelativeKey: strings.TrimPrefix(attrs.Name, prefix),
+			RelativeKey: strings.TrimPrefix(attrs.Name, basePath),
```
Comment (itaiad200) [inline pkg/ingest/store/gcs.go:57]: Are we okay with this being empty?

**Review by itaiad200 — CHANGES_REQUESTED** [review]

File: pkg/ingest/store/gcs.go (hunk 2 above)
Comment (nopcoder) [reply pkg/ingest/store/gcs.go:57]: empty string does nothing to the original value

File: pkg/ingest/store/gcs.go (hunk 1 above)
Comment (nopcoder) [reply pkg/ingest/store/gcs.go:28]: it supports the case of a single file and directory

**Review by itaiad200 — APPROVED** [review]
I get why it works for the bugs we have opened. I'm afraid of 2 things:
1. Regression in some other flow. It doesn't seem like our ingest/import tests cover the slashes bug well enough.
2. Inconsistency with other block adapters. Minor, but still.

Approving to unblock, please consider addressing these in followup issues.

**Comment by nopcoder** [comment]
Thanks @itaiad200 - tested from the UI and lakectl

- The above is based on our S3 walker implementation.
- Want to add test to cover this change.

**Pull request merged**